scripts/tests/chiptest/test_definition.py

```diff
@@ -142,30 +145,42 @@ def __startServer(self):
                     self.kvsPathSet.add(value)
         return self.runner.RunSubprocess(app_cmd, name='APP ', wait=False)
 
-    def __waitFor(self, waitForString, server_process, outpipe, timeoutInSeconds=10):
-        logging.debug('Waiting for %s' % waitForString)
+    def __waitFor(self, *symptoms, **kwargs):
+        """
+        Wait for all provided symptom strings to appear in the capture log.
+        """
+        server_process = kwargs.get('server_process', self.process)
+        outpipe = kwargs.get('outpipe', self.outpipe)
+        timeoutInSeconds = kwargs.get('timeoutInSeconds', 10)
+
+        logging.debug('Waiting for symptoms %r', symptoms)
 
         start_time = time.monotonic()
-        ready, self.lastLogIndex = outpipe.CapturedLogContains(
-            waitForString, self.lastLogIndex)
-        if ready:
-            self.lastLogIndex += 1
+        symptom_dict = {}
+
+        def searchSymptoms():
+            for s in symptoms:
```

Comment:
We seem to iterate over everything anyway. Can we re-write this to be easier to follow (less dicts and list compression) by the logic of `if first not found, then we did not find`? Thinking something like (untested):

```python
def logHasAllExpectations():
   maxLogIndex = self.lastLogIndex
   for s in expectations:
        found, index = outpipe.CapturedLogContains(s, self.lastLogIndex)
        if not found:
            return False
        maxLogIndex = max(maxLogIndex, index)
  self.lastLogIndex = maxLogIndex + 1
  return True
```